core/common/src/main/java/alluxio/util/network/NettyUtils.java

```diff
@@ -66,9 +69,16 @@ public static EventLoopGroup createEventLoop(ChannelType type, int numThreads,
    * {@link ChannelType}.
    *
    * @param type Selector for which form of low-level IO we should use
-   * @return ServerSocketChannel matching the ChannelType
+   * @param isDomainSocket whether this is a domain socket server
+   * @return ServerSocketChannel matching the requirements
    */
-  public static Class<? extends ServerChannel> getServerChannelClass(ChannelType type) {
+  public static Class<? extends ServerChannel> getServerChannelClass(ChannelType type,
+      boolean isDomainSocket) {
+    if (isDomainSocket) {
+      Preconditions.checkState(type == ChannelType.EPOLL,
+          "Domain socket is used while the channel type is not epoll.");
```

Comment:
Would "Domain sockets are only supported with EPOLL channel type" be more clear?